src/uucore/src/lib/features/proc_info.rs

```diff
@@ -237,6 +240,43 @@ impl ProcessInformation {
         Ok(time)
     }
 
+    pub fn ppid(&mut self) -> Result<u64, io::Error> {
+        // the PPID is the fourth field in /proc/<PID>/stat
+        // (https://www.kernel.org/doc/html/latest/filesystems/proc.html#id10)
+        self.stat()
+            .get(3)
+            .ok_or(io::ErrorKind::InvalidData)?
+            .parse::<u64>()
+            .map_err(|_| io::ErrorKind::InvalidData.into())
+    }
+
+    fn get_uid_or_gid_field(&mut self, field: &str, index: usize) -> Result<u32, io::Error> {
```

Comment:
What about using an enum for field? It would improve the checks of the compiler

Reply:
Not sure if worth it here given this is non public function with only callers right below it and unit tests will catch obvious typos

Reply:
Still. We don't know how this will evolve in the future :)